code/modules/paperwork/paper.dm

```diff
@@ -298,6 +311,46 @@
 
 		user << "<span class='notice'>You stamp the paper with your rubber stamp.</span>"
 
+
+	var/list/book_burners = list(/obj/item/weapon/weldingtool, /obj/item/weapon/lighter, /obj/item/weapon/match)
+	if(is_type_in_list(P, book_burners))

```

Comment:
I think you can just do if(is_hot(P)) to test if it'll light things, and that includes anything that light a cigarette, so burning paper with your cigarettes, or matches, or igniters, or any number of things is possible.